File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/SanPhamColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SanPhamRow } from "./SanPhamRow";

export interface SanPhamColumns {
    MaSp: Column<SanPhamRow>;
    HienThi: Column<SanPhamRow>;
    TenSp: Column<SanPhamRow>;
    GiaBan: Column<SanPhamRow>;
    DanhMucSanPham: Column<SanPhamRow>;
    TenNhaCungCap: Column<SanPhamRow>;
    TenThuongHieu: Column<SanPhamRow>;
    SoLuong: Column<SanPhamRow>;
    MoTa: Column<SanPhamRow>;
    GiamGia: Column<SanPhamRow>;
    GiaKhuyenMai: Column<SanPhamRow>;
    AnhSp: Column<SanPhamRow>;
    AnhSpCt: Column<SanPhamRow>;
    CreatedAt: Column<SanPhamRow>;
    UpdatedAt: Column<SanPhamRow>;
}

export class SanPhamColumns extends ColumnsBase<SanPhamRow> {
    static readonly columnsKey = 'SanPhamDB.SanPham';
    static readonly Fields = fieldsProxy<SanPhamColumns>();
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/SanPham/SanPhamColumns.cs
namespace wheystore.SanPhamDB.Columns;
using Serenity.ComponentModel;
using System;

[ColumnsScript("SanPhamDB.SanPham")]
[BasedOnRow(typeof(SanPhamRow), CheckNames = true)]
public class SanPhamColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MaSp { get; set; }
    [EditLink]
    public bool HienThi { get; set; }
    public string TenSp { get; set; }
    public decimal GiaBan { get; set; }

    [DisplayName("Danh Mục")]
    public string DanhMucSanPham { get; set; }

    [DisplayName("Nhà Cung Cấp")]
    public string TenNhaCungCap { get; set; }

    [DisplayName("Thương Hiệu")]
    public string TenThuongHieu { get; set; }

    [DisplayName("Trạng Thái"),
    BooleanFormatter(TrueText = "Đang giảm giá", FalseText = "Không giảm")]

    public int SoLuong { get; set; }
    public string MoTa { get; set; }

    public decimal GiamGia { get; set; }
    public decimal GiaKhuyenMai { get; set; }
   


    public string AnhSp { get; set; }
    public string AnhSpCt { get; set; }
   
    
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    

    
}